crates/services/bls_aggregation/src/lib.rs

```diff
@@ -1,7 +1,367 @@
+//! # use eigen_logging::get_test_logger;
+//! # use eigen_testing_utils::{
+//! #     anvil_constants::{
+//! #         get_operator_state_retriever_address, get_registry_coordinator_address,
+//! #     },
+//! # };
+//! # use eigen_services_operatorsinfo::operatorsinfo_inmemory::OperatorInfoServiceInMemory;
+//! # use eigen_testing_utils::anvil_constants::{ANVIL_HTTP_URL, ANVIL_WS_URL};
+//! # #[tokio::main]
+//! # async fn main() {
+//! #     let http_endpoint = ANVIL_HTTP_URL;
+//! #     let ws_endpoint = ANVIL_WS_URL;
+//! #     let logger = get_test_logger();
+//! #     let registry_coordinator_address =
+//! #         get_registry_coordinator_address(http_endpoint.to_string()).await;
+//! #     let operator_state_retriever_address =
+//! #         get_operator_state_retriever_address(http_endpoint.to_string()).await;
+//! #     
+//! #     // Create avs clients to interact with contracts deployed on anvil
+//! #     let avs_registry_reader = AvsRegistryChainReader::new(
+//! #         get_test_logger(),
+//! #         registry_coordinator_address,
+//! #         operator_state_retriever_address,
+//! #         http_endpoint.to_string(),
+//! #     )
+//! #     .await
+//! #     .unwrap();
+//! #
+//! #     // Create operators info service
+//! #     let operators_info = OperatorInfoServiceInMemory::new(
+//! #         get_test_logger(),
+//! #         avs_registry_reader.clone(),
+//! #         ws_endpoint.to_string(),
+//! #     )
+//! #     .await
+//! #     .unwrap()
+//! #     .0;
+//! #
+//! #     // Create avs registry service chain caller
+//! #     let avs_registry_service =
+//! #         AvsRegistryServiceChainCaller::new(avs_registry_reader.clone(), operators_info);
+//! #    let (service_handle, mut aggregate_receiver) =
+//! #        BlsAggregatorService::new(avs_registry_service, logger).start();
+//! #
+//!     match aggregate_receiver.receive_aggregated_response().await {
+//!         Ok(aggregated_response) => {
+//!             // Handle the aggregated response
+//!         }
+//!         Err(e) => {
+//!             // Handle the error
+//!         }
+//!     }
+//! # }
+//! ```
+//!
+//! ## Example Diagram
+//!
+//! The following diagram shows the sequence of events when a user creates the BLS Aggregator Service, starts the service, and then initializes a task. The service processes two signatures from the operators and then aggregates them into a single aggregated signature.
+//!
+//! <pre>
+#![cfg_attr(doc, doc = simple_mermaid::mermaid!("../diagram/sequence-bls.mmd"))]
+//! </pre>
+//!
+//! ## Testing
+//!
+//! To run the [integration tests](https://github.com/Layr-Labs/eigensdk-rs/blob/dev/crates/services/bls_aggregation/src/bls_agg_test.rs), you can use the following command:
+//!
+//! ```sh
+//! cargo test --package eigen-services-blsaggregation --lib -- bls_agg_test::integration_test --show-output
+//! ```
+//! [`new()`]: bls_agg::BlsAggregatorService::new()
```

Comment:
Unrelated, but I think we should move the `BlsAggregatorService` to this file (or reexport it here).